src/snowflake/snowpark/mock/file_operation.py

```diff
@@ -0,0 +1,178 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+import os
+from collections import defaultdict
+from functools import partial
+from logging import getLogger
+from typing import TYPE_CHECKING, Dict, List
+
+import pandas as pd
+
+from snowflake.snowpark._internal.analyzer.expression import Attribute
+from snowflake.snowpark.exceptions import SnowparkSQLException
+from snowflake.snowpark.mock.snowflake_data_type import ColumnEmulator, TableEmulator
+from snowflake.snowpark.mock.snowflake_to_pandas_converter import CONVERT_MAP
+from snowflake.snowpark.types import DecimalType, StringType
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.mock.mock_analyzer import MockAnalyzer
+
+
+_logger = getLogger(__name__)
+
+# this map keeps the information for remote file to local file,
+# key is the remote 'stage/file', value is the local file path
+_FILE_STAGE_MAP = {}
+
+# this map keeps the information the files each stage stores
+# key is the remote 'stage', value is the set of file names stored in the stage
+_STAGE_FILE_MAP = defaultdict(set)
+
+PUT_RESULT_KEYS = [
+    "source",
+    "target",
+    "source_size",
+    "target_size",
+    "source_compression",
+    "target_compression",
+    "status",
+    "message",
+]
+
+SUPPORTED_CSV_READ_OPTIONS = (
+    "SKIP_HEADER",
+    "SKIP_BLANK_LINES",
+    "FIELD_DELIMITER",
+    "FIELD_OPTIONALLY_ENCLOSED_BY",
+)
+
+
+def put(local_file_name: str, stage_location: str) -> TableEmulator:
+    """
+    Put a file into in memory map, key being stage location and value being the local file path
+    """
+    local_file_name = local_file_name[
+        8:-1
+    ]  # skip normalized prefix `file:// and suffix `
+    file_name = os.path.basename(local_file_name)
+    remote_file_path = f"{stage_location}/{file_name}"
+    _FILE_STAGE_MAP[remote_file_path] = local_file_name
+    _STAGE_FILE_MAP[f"{stage_location}/"].add(local_file_name)
+    file_size = os.path.getsize(os.path.expanduser(local_file_name))
+
+    result_df = TableEmulator(
+        columns=PUT_RESULT_KEYS,
+        sf_types={
+            "source": StringType(),
+            "target": StringType(),
+            "source_size": DecimalType(10, 0),
+            "target_size": DecimalType(10, 0),
+            "source_compression": StringType(),
+            "target_compression": StringType(),
+            "status": StringType(),
+            "message": StringType(),
+        },
+        dtype=object,
+    )
+    result_df.loc[len(result_df)] = {
+        k: v
+        for k, v in zip(
+            PUT_RESULT_KEYS,
+            [file_name, file_name, file_size, file_size, None, None, None, None],
+        )
+    }
+    return result_df
+
+
+def read_file(
```

Comment:
Should this be in a different mock file? `file_operation` only does read/write, but does not try to understand the file.

Reply:
for now let's just put them in the same module, I think we can split in the future